lnbits/wallets/alby.py

```diff
@@ -0,0 +1,126 @@
+        self.auth = {"Authorization": "Bearer " + settings.alby_api_key}
+        self.client = httpx.AsyncClient(base_url=self.endpoint, headers=self.auth)
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        try:
+            r = await self.client.get("/balance", timeout=10)
+        except (httpx.ConnectError, httpx.RequestError):
+            return StatusResponse(f"Unable to connect to '{self.endpoint}'", 0)
+
+        data = r.json()["balance"]
+        if r.is_error:
+            return StatusResponse(data["error"], 0)
+
+        return StatusResponse(None, data)
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        if description_hash or unhashed_description:
+            raise Unsupported("description_hash")
+
+        r = await self.client.post(
+            "/invoices",
+            json={
+                "amount": amount,
+                "description": memo or "",
+            },
+            timeout=40,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()
+        checking_id = data["payment_hash"]
+        payment_request = data["payment_request"]
+        return InvoiceResponse(True, checking_id, payment_request, None)
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        # https://api.getalby.com/payments/bolt11
+        r = await self.client.post(
+            "/payments/bolt11",
+            json={"invoice": bolt11},  # assume never need amount in body
+            timeout=None,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return PaymentResponse(False, None, None, None, error_message)
+
+        data = r.json()
+        checking_id = data["payment_hash"]
+        fee_msat = -data["fee"]
+        preimage = data["payment_preimage"]
+
+        return PaymentResponse(True, checking_id, fee_msat, preimage, None)
+
+    async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
+        return await self.get_payment_status(checking_id)
+
+    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
+        # Note from API: currently only settled Alby invoices can be retrieved.
+        r = await self.client.get(f"/invoices/{checking_id}")
+
+        if r.is_error:
+            return PaymentStatus(None)
+
+        data = r.json()
+
+        statuses = {
+            "CREATED": None,
+            "SETTLED": True,
+            "error": None,
```

Comment:
are "error" and "failed" actually possible statuses? they seem inconsistent with the Alby API

Reply:
Please make sure your docs are up to date on the Alby API and say what states are returned. 

Normally APIs do return an error or failed state.

Reply:
Will remove these here for now

Reply:
done